src/interdiff.c

```diff
@@ -58,6 +59,27 @@
 #define PATCH "patch"
 #endif
 
+#define COLOR_RESET     "\033[0m"
+
+/* Line type for coloring */
+enum line_type {
+	LINE_FILE = 0,
+	LINE_HEADER,
+	LINE_HUNK,
+	LINE_ADDED,
+	LINE_REMOVED,
+};
+
+/* ANSI color codes for diff output */
+static char*color_codes[] = {
```

Comment:
Also, initialize the array size to `LINE_MAX`, i.e. (`color_codes[]` ==> `color_codes[LINE_MAX]`).

This ensures any future changes made to `enum line_type` that accidentally lack a complementary change to the `color_codes` array won't cause an out-of-bounds read.